src/marionette.rs

```diff
@@ -334,11 +334,12 @@ impl MarionetteHandler {
                          .map_err(|_| WebDriverError::new(ErrorStatus::UnknownError,
                                                           "Unable to read profile preferences file")));
 
-        prefs.insert("marionette.defaultPrefs.port", Pref::new(port as i64));
+        for &(ref name, ref value) in prefs::DEFAULT.iter() {
```

Comment:
I guess this might be a result of https://github.com/mozilla/geckodriver/commit/2bfdc3ec8151c427a6a75a6ba3ad203459540495.

Many of the preferences in `prefs::DEFAULT` are actually required for Marionette to function correctly, and some of these we can’t set at runtime. We should probably move those that are absolute requirements over to `prefs::REQUIRED`, and we should set those we can set at runtime in Marionette. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1323290 about the latter before Christmas.

As I read this code, the default prefs will only get written to the profile if a custom profile is not provided. The intention of `prefs::DEFAULT` is to say “we recommend these settings for running Marionette”, and triggering another code path that doesn’t set these if the user provides a profile a bit surprising in my opinion.

I’m also curious what the reason for this change is. I guess you are providing a profile with some preferences that get overridden by geckodriver?